### ISSUE TITLE
.Net Standard 2.1 and org.nuget.system.runtime.compilerservices.unsafe

### ISSUE THREAD

**Issue by favoyang** [xoofx/UnityNuGet#68]
I noticed some packages (maybe all) have been added NET_STANDARD_2_0 in Define Constraints. This is good for packages like `system.memory` to avoid conflict with .a Net Standard 2.1 profile which shipped with Unity 2021.2.

![image](https://user-images.githubusercontent.com/125390/139689255-21392d08-c8c6-4ebd-95c9-3973179fb4b7.png)

However, today I tracked a package `org.nuget.system.runtime.compilerservices.unsafe`, which has been removed from the unity profiles
- https://issuetracker.unity3d.com/issues/using-a-net-standard-2-dot-1-dll-that-references-system-dot-runtime-dot-compilerservices-dot-unsafe-via-nuget-results-in-missingmethodexception
- https://github.com/Unity-Technologies/mono/pull/1486

Seems for `org.nuget.system.runtime.compilerservices.unsafe` it needs to remove the NET_STANDARD_2_0 to make it both work with Net Standard 2.0 and Net Standard 2.1.

I also noticed https://github.com/xoofx/UnityNuGet/issues/52, suggesting removing NET_STANDARD_2_0  from all packages, this shouldn't be applied to packages like `system.memory` and perhaps other packages that are not compatible with Unity's .net standard 2,1 profile.

**Comment by xoofx** [comment]
> I noticed some packages (maybe all) have been added NET_STANDARD_2_0 in Define Constraints. This is good for packages like system.memory to avoid conflict with .a Net Standard 2.1 profile which shipped with Unity 2021.2.

Yes, it was the previous version of UnityNuGet. You have to remove your cache folder `%localappdata%\Unity\cache\npm\unitynuget-registry.azurewebsites.net)` in order to get the new defines.

> Seems for org.nuget.system.runtime.compilerservices.unsafe it needs to remove the NET_STANDARD_2_0 to make it both work with Net Standard 2.0 and Net Standard 2.1.

In the new version `system.runtime.compilerservices.unsafe` should be included only for Unity versions before 2021.2, otherwise it will use the version that ships with Unity 2021.2 (and not the one coming from the package). Though I can see that it is not correctly setting the define constraints for this, I will have to double check

**Comment by favoyang** [comment]
Thanks for the cache tips. After clearing the cache. I confirmed that the rebuilt `system.runtime.compilerservices.unsafe` has an empty Define Constraints. Then, it can be loaded by both Unity 2021.2 (with net standard 2.1) and previous Unity versions (with net standard 2.0).
![image](https://user-images.githubusercontent.com/125390/139710948-25714057-30c7-49ec-9b8c-60b81e594be3.png)

> otherwise it will use the version that ships with Unity 2021.2

From the Unity issue tracker, `system.runtime.compilerservices.unsafe` has been removed from the ".net standard 2.1" profile by Unity.
https://issuetracker.unity3d.com/issues/using-a-net-standard-2-dot-1-dll-that-references-system-dot-runtime-dot-compilerservices-dot-unsafe-via-nuget-results-in-missingmethodexception
